### PR TITLE
implements querystring transformation

### PR DESCRIPTION
**Pull request by jessopb**
querystrings containing h=#&w=# will scale the image.
querystrings containing an additional t=crop will crop to the center
system supports addition of more transformations responding to querystring vales as private functions inside transformImage.js

### CODE REVIEW

File: server/controllers/assets/utils/serveFile.js [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
+  } catch (e) {}
 
-const serveFile = ({ filePath, fileType }, res) => {
   if (!fileType) {
     logger.error(`no fileType provided for ${filePath}`);
   }
+
+  let mediaType = fileType ? fileType.substr(0, fileType.indexOf('/')) : '';
+  const transform =
+    mediaType === 'image' && queryObject.hasOwnProperty('h') && queryObject.hasOwnProperty('w');
```
Comment (skhameneh) [inline server/controllers/assets/utils/serveFile.js]: Can format like:
```
  const transform = (
    mediaType === 'image' && queryObject.hasOwnProperty('h') && queryObject.hasOwnProperty('w')
  );
```

File: server/controllers/assets/utils/transformImage.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const gm = require('gm');
+const imageMagick = gm.subClass({ imageMagick: true });
+const { getImageHeightAndWidth } = require('../../../utils/imageProcessing');
+
+module.exports = function transformImage(path, queryObj) {
+  return new Promise((resolve, reject) => {
+    let cHeight = queryObj.h || null;
+    let cWidth = queryObj.w || null;
+    let transform = queryObj.t || null;
```
Comment (skhameneh) [inline server/controllers/assets/utils/transformImage.js]: Can use a destructuring assignment, see https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/

File: server/controllers/assets/utils/transformImage.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    let oHeight,
+      oWidth = null;
+
+    getImageHeightAndWidth(path).then(hwarr => {
```
Comment (skhameneh) [inline server/controllers/assets/utils/transformImage.js]: Can use `await`

**Comment by jessopb** [comment]
@skhameneh Give it one more look, will ya? I modified the function to resize to match wordpress behavior, and added a stretch function.

File: server/controllers/assets/utils/transformImage.js (hunk 3 above)
Comment (jessopb) [reply server/controllers/assets/utils/transformImage.js]: I agree, but that's enough time spent on this feature for now.

**Pull request merged**